### ISSUE TITLE
Passing a string with a % to Text() causes garbage data to be printed

### ISSUE THREAD

**Issue by shayded-exe** [ImGuiNET/ImGui.NET#442]
![image](https://github.com/ImGuiNET/ImGui.NET/assets/1417838/bb230462-3ff3-4a6e-89bb-1ccfb7209e91)

![image](https://github.com/ImGuiNET/ImGui.NET/assets/1417838/1f315794-0418-416f-937d-45e7b5379c9a)

[`ImFormatStringV`](https://github.com/ocornut/imgui/blob/2a6d7b1eaa1851e5b53d8c7a361a8d54f5a9ae8c/imgui.cpp#L1970)

If `vsnprintf` interprets the string passed to `Text` as a format string, it looks at `va_list args` which will always be empty. The result is the next pointer being formatted into the string.

ImGui.NET should allow for passing additional args to `Text` so I don't have to pre-format my strings.

**Comment by shayded-exe** [comment]
I did just realize that a workaround is to escape the `%` with an additional `%` like this:
`var str = $"{value:p1}% - {label}";`

**Comment by zaafar** [comment]
yeah, that's how it is, if you wanna read more about it there was a recent discussion about it over here.
https://github.com/dearimgui/dear_bindings/issues/45

**Comment by shayded-exe** [comment]
Ah I see.

Not a big deal since it can easily be worked around, but it would be nice if this issue had higher visibility so future people like me don't spend a couple hours in confusion :)

Maybe I could do a PR to start adding a "known gotchas" section to the docs to highlight oddities like this.

**Comment by RikkiGibson** [comment]
It looks like dearimgui/dear_bindings#46 adds the ability to generate a C header with unformatted variants of all the format string consuming functions. Would it be possible for ImGui.NET to adopt that, and just use unformatted versions of all relevant methods across the board?

**Comment by sonoro1234** [comment]
Is `CIMGUI_VARGS0` useful?
https://github.com/cimgui/cimgui/pull/297

**Comment by RikkiGibson** [comment]
https://github.com/cimgui/cimgui/pull/297#issuecomment-2887383643 looks promising, it sounds like this project could just define that symbol when generating the header, and start referring to the non variadic versions of the functions, and things would work?

**Comment by sonoro1234** [comment]
That should be.
